src/mvp/objects/Animator.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "Base.hpp"
+#include "GameData.hpp"
+#include "Renderer.hpp"
+#include "TreasureIsland.hpp"
+#include <vector>
+
+namespace admirals::mvp::objects {
+
+enum SpriteType {
+    ShipType,
+    Island,
+    none,
+};
+
+class Animator : public GameObject {
```

Comment:
I would prefer a more general non-GameObject animator, but this works, although maybe we shouldn't make this static and instead add a reference to the animator in `GameData`.

Reply:
I could implement the function in renderer instead and let it handle the update of the frames